### ISSUE TITLE
`.map(|x| x.into())` → `.map(Into::into)`

### ISSUE THREAD

**Issue by fenollp** [rust-lang/rust-clippy#10440]
### What it does

This lint replaces the introduction of a simple closure with the named function that is being used in said closure.

### Lint Name

replace_mapped_closure_with_into

### Category

complexity

### Advantage

- Reduces complexity of the `map` by introducing a common function name instead of a closure + identifier

### Drawbacks

* May be less readable for newcomers.
* No performance drawback.

### Example

```rust
impl IpResource {
    pub fn mac_address(&self) -> Option<MacAddr6> {
        self.mac_address.map(|x| x.into())
    }
}
```

Could be written as:

```rust
impl IpResource {
    pub fn mac_address(&self) -> Option<MacAddr6> {
        self.mac_address.map(Into::into)
    }
}
```

**Comment by Alexendoo** [comment]
There's an existing lint that covers this: [`redundant_closure_for_method_calls`](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure_for_method_calls)

It is in the `pedantic` category though so it doesn't warn by default - #4101

**Comment by samueltardieu** [comment]
Also note that, unlike in many functional languages, `|x| f(x)` cannot always be replaced by `f` (η-reduction) in Rust as `f(x)` might involve one or more reference and one dereference of `x` (such as `&***x`). Those dereferences/reference won't happen automatically when using only the function name, which wants the exact signature as declared.
